cirq/ops/raw_types.py

```diff
@@ -459,11 +466,18 @@ def controlled_by(self,
                 default to 1.
         """
         # Avoids circular import.
-        from cirq.ops import ControlledOperation
+        from cirq.ops.controlled_operation import ControlledOperation
         if len(control_qubits) == 0:
             return self
         return ControlledOperation(control_qubits, self, control_values)
 
+    def with_probability(self, probability: Union[float, sympy.Basic, float]):
```

Comment:
shouldn't this be on GateOperation and not on the general Operation?

Reply:
No, any operation can specify an underlying gate.